src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php

```diff
@@ -60,4 +69,40 @@ public function theProductOriginalPriceShouldBe(int $originalPrice): void
 
         Assert::same($response['originalPrice'], $originalPrice);
     }
+
+    /**
+     * @Then /^I should see ("[^"]+" variant) is discounted from ("[^"]+") to ("[^"]+") with "([^"]+)" promotion$/
+     */
+    public function iShouldSeeVariantIsDiscountedFromToWithPromotion(
+        ProductVariantInterface $targetVariant,
+        int $originalPrice,
+        int $price,
+        string $promotionName
+    ): void {
+        $response = $this->responseChecker->getResponseContent($this->client->getLastResponse());
+
+        foreach ($response['hydra:member'] as $variant) {
+            if ($variant['code'] !== $targetVariant->getCode()) {
+                continue;
+            }
+
+            Assert::same($variant['price'], $price);
+            Assert::same($variant['originalPrice'], $originalPrice);
+            Assert::inArray(['name' => $promotionName], $variant['appliedPromotions']);
+        }
+    }
+
+    /**
+     * @Then I should see :firstVariant variant and :secondVariant variant are not discounted
+     */
+    public function iShouldSeeVariantsAreNotDiscounted(string ...$variantsNames): void
+    {
+        $response = $this->responseChecker->getResponseContent($this->client->getLastResponse());
```

Comment:
can we use here `$this->responseChecker->getCollection($this->client->getLastResponse());` ?